### PR TITLE
Tiny: allow CaseExpression with operand and else clause

### PR DESCRIPTION
**Pull request by roji**

### CODE REVIEW

**Comment by smitpatel** [comment]
It does not make sense. What is the Sql expression tree you are constructing?

**Comment by roji** [comment]
It's supported in SQL both by [SQL Server](https://docs.microsoft.com/en-us/sql/t-sql/language-elements/case-transact-sql?view=sql-server-2017) (the "simple case expression") and [PostgreSQL](https://www.postgresql.org/docs/current/functions-conditional.html) (again "simple form").

The specific translation where I could use this is to return a number, but to translate 0 into 7 (this actually happens in PostgreSQL when translating DayOfWeek which returns an enum). I could generate `CASE value WHEN 0 THEN 7 ELSE value`.

File: src/EFCore.Relational/Query/Pipeline/SqlExpressionFactory.cs [hunk 1]
```diff
@@ -271,12 +271,15 @@ public SqlUnaryExpression Not(SqlExpression operand)
         public SqlUnaryExpression Negate(SqlExpression operand)
             => MakeUnary(ExpressionType.Negate, operand, operand.Type, operand.TypeMapping);
 
-        public CaseExpression Case(SqlExpression operand, params CaseWhenClause[] whenClauses)
+        public CaseExpression Case(SqlExpression operand, SqlExpression elseResult, params CaseWhenClause[] whenClauses)
         {
             var operandTypeMapping = operand.TypeMapping
                 ?? whenClauses.Select(wc => wc.Test.TypeMapping).FirstOrDefault(t => t != null)
+                ?? elseResult?.TypeMapping
```
Comment (smitpatel) [inline src/EFCore.Relational/Query/Pipeline/SqlExpressionFactory.cs]: This is incorrect. You can do
```
Case value
   When 1 then "1"
   Else "NULL"
End
```

Comment (roji) [reply src/EFCore.Relational/Query/Pipeline/SqlExpressionFactory.cs]: Of course you're right, thanks.

**Pull request merged**